### PR TITLE
Feature/support multi instance types

### PR DESCRIPTION
**Pull request by TalentNing**
=== RUN   TestAccAlicloudEssScalingConfiguration_basic
--- PASS: TestAccAlicloudEssScalingConfiguration_basic (78.95s)
=== RUN   TestAccAlicloudEssScalingConfiguration_SystemDisk
--- PASS: TestAccAlicloudEssScalingConfiguration_SystemDisk (70.43s)
=== RUN   TestAccAlicloudEssScalingConfiguration_multiConfig
--- PASS: TestAccAlicloudEssScalingConfiguration_multiConfig (77.60s)
=== RUN   TestAccAlicloudEssScalingConfiguration_active
--- PASS: TestAccAlicloudEssScalingConfiguration_active (70.10s)
=== RUN   TestAccAlicloudEssScalingConfiguration_disable
--- PASS: TestAccAlicloudEssScalingConfiguration_disable (130.87s)
=== RUN   TestAccAlicloudEssScalingConfiguration_modify
--- PASS: TestAccAlicloudEssScalingConfiguration_modify (122.55s)
=== RUN   TestAccAlicloudEssScalingConfiguration_multi_sgs
--- PASS: TestAccAlicloudEssScalingConfiguration_multi_sgs (72.62s)
=== RUN   TestAccAlicloudEssScalingConfiguration_modify_single_2_multi_sg
--- PASS: TestAccAlicloudEssScalingConfiguration_modify_single_2_multi_sg (100.13s)
=== RUN   TestAccAlicloudEssScalingConfiguration_multi_instance_types
--- PASS: TestAccAlicloudEssScalingConfiguration_multi_instance_types (74.82s)
=== RUN   TestAccAlicloudEssScalingConfiguration_modify_single_2_multi_instance_types
--- PASS: TestAccAlicloudEssScalingConfiguration_modify_single_2_multi_instance_types (95.18s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	893.313s

### CODE REVIEW

File: alicloud/resource_alicloud_ess_scalingconfiguration.go [hunk 1]
```diff
@@ -286,11 +288,22 @@ func modifyEssScalingConfiguration(d *schema.ResourceData, meta interface{}) err
...
-		instanceTypes = append(instanceTypes, d.Get("instance_type").(string))
-		args.InstanceTypes = &instanceTypes
-		d.SetPartial("instance_type")
+	hasChangeInstanceType := d.HasChange("instance_type")
+	hasChangeInstanceTypes := d.HasChange("instance_types")
+	if hasChangeInstanceType || hasChangeInstanceTypes {
+		instanceType := d.Get("instance_type").(string)
+		instanceTypes := d.Get("instance_types").([]interface{})
+		if instanceType == "" && (instanceTypes == nil || len(instanceTypes) == 0) {
+			return fmt.Errorf("securityGroupId or securityGroupIds must be assigned")
```
Comment (YY-tomorrow) [inline alicloud/resource_alicloud_ess_scalingconfiguration.go]: this should be instance_type or instance_types must be assigned

File: website/docs/r/ess_scaling_configuration.html.markdown [hunk 2]
```diff
@@ -40,7 +40,8 @@ The following arguments are supported:
 
 * `scaling_group_id` - (Required, ForceNew) ID of the scaling group of a scaling configuration.
 * `image_id` - (Required) ID of an image file, indicating the image resource selected when an instance is enabled.
-* `instance_type` - (Required) Resource type of an ECS instance.
+* `instance_type` - (Optional) Resource type of an ECS instance.
+* `instance_types` - (Optional) Resource types of an ECS instance.
```
Comment (YY-tomorrow) [inline website/docs/r/ess_scaling_configuration.html.markdown]: please add (Available in 1.46.0)

File: alicloud/resource_alicloud_ess_scalingconfiguration.go (hunk 1 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_scalingconfiguration.go]: fixed

File: website/docs/r/ess_scaling_configuration.html.markdown (hunk 2 above)
Comment (TalentNing) [reply website/docs/r/ess_scaling_configuration.html.markdown]: done

**Pull request merged**